src/js/components/form-elements/v2/DateField.jsx

```diff
@@ -25,17 +25,27 @@ const DateField = ({
   className,
   value,
   onChange,
+  showTimeSelect,
   ...fieldProps
 }) => {
   const translate = useTranslate();
   const onClear = () => onChange(null);
 
-  const onChangeHandler = date => onChange(date?.format(DateFormat.MMM_DD_YYYY));
+  const onChangeHandler = (date) => {
+    if (showTimeSelect) {
+      onChange(date?.format(DateFormat.MMM_DD_YYYY_HH_MM_SS));
+      return;
+    }
+    onChange(date?.format(DateFormat.MMM_DD_YYYY));
+  };
 
   const formatDate = (dateToFormat) => {
     if (!dateToFormat) {
       return null;
     }
+    if (showTimeSelect) {
+      return moment(new Date(dateToFormat), DateFormat.MMM_DD_YYYY_HH_MM_SS);
+    }
 
     return moment(new Date(dateToFormat), DateFormat.MMM_DD_YYYY);
```

Comment:
It can be reduced using ternary operator

Reply:
I would say it doesn't reduce much, and it's just a matter of preference, as it's a simple one if statement. 
Anyway, changed to ternary.